### ISSUE TITLE
Weird lazarus tile

### ISSUE THREAD

**Issue by qndel** [diasurgical/devilutionX#1970]
When lazarus is "locked", there's still 1 tile inside the locked area you can teleport to

![image](https://user-images.githubusercontent.com/14297035/118310467-f1fc6c00-b4ee-11eb-9dbf-14a1eb016b41.png)

**Comment by nickgavran** [comment]
This is because they used the wrong tile. There are more spots like this, most notably if you go southeast, near the door you can teleport out there, there's a whole row of it.

**Comment by AJenbo** [comment]
Do you happen to know the tile id?

**Comment by julealgon** [comment]
Can we use @glebm tile overlay to spot these issues?

**Comment by AJenbo** [comment]
Simpler to just load the level in Tiled and check the tiles

**Comment by nickgavran** [comment]
This tile is the one used in many places in Lazarus' place (Too many to show), which is a Roof+Ground tile, and it allows for players/monsters to stand on it (4th last tile in the Cathedral tileset):
![RoofPlusGroundTile](https://user-images.githubusercontent.com/60329165/118412525-d766f600-b69a-11eb-98c5-a7a75e35d406.png)

This is the correct tile that it should use instead, a simple Roof tile: 
![NormalRoofTile](https://user-images.githubusercontent.com/60329165/118412578-1137fc80-b69b-11eb-8a51-47cd755177fe.png)

They seem to use these interchangably in this specific dungeon, I am unsure as to why.

**Comment by nickgavran** [comment]
This is the tile which is causing the problems: - This is the 4th last tile in the Cathedral tileset, and it is both a roof and a ground tile in one (thus you can stand in it). They scattered these around in this specific dungeon (too many to show), I am unsure as to why.
![RoofandGround](https://user-images.githubusercontent.com/60329165/118412687-ab984000-b69b-11eb-9152-fe0a4ca758f5.png)

It should be using this tile instead, a regular roof only tile: 
![NormalRoof](https://user-images.githubusercontent.com/60329165/118412721-da161b00-b69b-11eb-8c5b-e14fc576cc54.png)

**Comment by AJenbo** [comment]
Afaik the tile is intended  for padding the outer edge of the map. If we change the rendering to render ground roof tiles in the void we can replace all instance of it.

**Comment by julealgon** [comment]
> and it is both a roof and a ground tile in one (thus you can stand in it)

This is so confusing to me. Why would such tile exist? Are there any valid instances where you want to be able to stand on a "roof tile" in the normal game?

**Comment by AJenbo** [comment]
> This is so confusing to me. Why would such tile exist?

For some reason they felt it was a simpler solution then to have the code handle it. They render seamed pretty locked in place early one so maybe they didn't want to touch the 10k ASM mess and went with this to cover the edge of the map.

> Are there any valid instances where you want to be able to stand on a "roof tile" in the normal game?

Most likely they forgot to mark it as solid in the solid table (`Levels\L1Data\L1.SOL`). There are lots of easy ways to improve this.

**Comment by nickgavran** [comment]
> > and it is both a roof and a ground tile in one (thus you can stand in it)
> 
> This is so confusing to me. Why would such tile exist? Are there any valid instances where you want to be able to stand on a "roof tile" in the normal game?

Yes. It is used in the 3 tiles behind the entrance in the structure leading to Skeleton King's Lair. Behind the doorway, you can see the ground there. If you use regular roof tiles in that spot, you can see the void. This is the only instance I know of it existing properly in the game.

**Comment by julealgon** [comment]
> > > and it is both a roof and a ground tile in one (thus you can stand in it)
> > 
> > 
> > This is so confusing to me. Why would such tile exist? Are there any valid instances where you want to be able to stand on a "roof tile" in the normal game?
> 
> Yes. It is used in the 3 tiles behind the entrance in the structure leading to Skeleton King's Lair. Behind the doorway, you can see the ground there. If you use regular roof tiles in that spot, you can see the void. This is the only instance I know of it existing properly in the game. However, why you can stand on it is dumb, they should have just made it solid and kept it for the aesthetics where it is needed.

It's funny you mention that... @nitekat 's bot struggled with this recently because a gold pile was dropped in that tile (behind the trigger to go down to leoric's tomb) and the bot got stuck trying to grab the gold: every time he tried to grab it, he inadvertently entered the tomb instead, creating an endless cycle.
